pkgs/development/lua-modules/generated-packages.nix

```diff
@@ -1618,8 +1618,8 @@ plenary-nvim = buildLuarocksPackage {
   version = "scm-1";
 
   knownRockspec = (fetchurl {
-    url    = "https://luarocks.org/plenary.nvim-scm-1.rockspec";
-    sha256 = "1xgqq0skg3vxahlnh1libc5dvhafp11k6k8cs65jcr9sw6xjycwh";
+    url    = "https://raw.githubusercontent.com/nvim-lua/plenary.nvim/master/plenary.nvim-scm-1.rockspec";
```

Comment:
it shouldn't be fetched directly from master, specify a revision

Reply:
I just followed how other packages are doing it. It's important to note that the commit reference below doesn't have an up-to-date rockspec (it got out of sync, the next commit - currently master - fixes this).

Reply:
if upstream gets another update on this file, this `fetchurl` would break because `sha256` wouldn't match

Reply:
I haven't looked but I guess the `update-luarocks-packages` script which generates this file is supposed to deal with this. Every single other occurrence of `githubusercontent.com` in the file points to `master` too. I guess it's an artifact of the update script and should be fixed in the way you suggest.

Reply:
These are immutable from a mirror repository https://github.com/NixOS/nixpkgs/blob/f9137f6131091389fac53df9cee02a45b1d4ad49/pkgs/build-support/fetchurl/mirrors.nix#L13